elastic/kibana
# [Lens][Embeddable] Fix infinite rendering loop for custom component

## Summary

Fixes #202266 

This is a fix for a regression bug due to the new Embeddable refactor.
The previous Embeddable system had a `fetch$` wrapper able to distill changes from the unified search api.

The new refactor did miss to watch the correct unified search API to fetch, and this produced issues, in particular when the custom Lens component is re-rendered due to other state changes: in fact new search context parameter could be sent causing an infinite rendering loop as nothing was stopping to repeat the same `fetch` over and over.

The final fix is about splitting the search context subscription into 2 parts:
* panel search context
* unified search context

Both will be watched for changes and reloaded only when required.
The `data_loader` module has been split into smaller modules with specific responsibilities now, and reload events are logger with more granularity.

I've added some FTR tests on the Rule flyout page to check for:
* multiple renderings
* the error message is correctly handled
* it can restore the chart after an error

## Notes for reviewers

To test this fix try the following:
* Create a new Rule within the Observability / Alerts page
* Go in Discover and test the `autoRefresh` feature
* Go in Discover and test that brushing the chart works fine
* Go in Dashboard and repeat the same step as in Discover
* Go in Dashboard and set a panel time range and check it updates correctly

### Checklist
